responses/fetch.go

```diff
@@ -31,6 +33,25 @@ func (r *Fetch) Handle(resp imap.Resp) error {
 		return err
 	}
 
+	if r.Uid && msg.Uid == 0 {
+		// we requested UIDs and got a message without --> unilateral update --> ignore
+		return ErrUnhandled
+	}
+
+	var num uint32
+	if r.Uid {
+		num = msg.Uid
+	} else {
+		num = seqNum
+	}
+
+	// check whether we obtained a result we requested with our SeqSet
+	// If it does not, but the msg contains a UID and the set is dynamic (i.e. * or n:*), the server
+	// supplied us with the max UID. That is fine.
```

Comment:
Hm. What do you mean by "the server supplied us with the max UID"?

Reply:
From the RFC, I got that when requesting a dynamic UID sequence like "4:*" the server can answer with the message with UID "2" iff this is the maximum UID. (Reasoning: "4:*" == "*:4")

Reply:
Ah. Indeed:

```
                    ; Example: a unique identifier sequence range of
                    ; 3291:* includes the UID of the last message in
                    ; the mailbox, even if that value is less than 3291.
```

Can you expand the comment a bit?

Reply:
I did (see fresh commit): Is this to your liking?

```go
// Check whether we obtained a result we requested with our SeqSet
// If the result is not contained in our SeqSet we have to handle an additional special case:
// In case we requested UIDs with a dynamic sequence (i.e. * or n:*) and the maximum UID of the mailbox
// is less than our n, the server will supply us with the max UID (cf. RFC 3501 §6.4.8 and §9 `seq-range`).
// Thus, such a result is correct and has to be returned by us.
```